x-pack/plugins/ingest_manager/server/services/setup.ts

```diff
@@ -9,16 +9,79 @@ import { CallESAsCurrentUser } from '../types';
 import { agentConfigService } from './agent_config';
 import { outputService } from './output';
 import { ensureInstalledDefaultPackages } from './epm/packages/install';
+import {
+  packageToConfigDatasourceInputs,
+  Datasource,
+  AgentConfig,
+  Installation,
+  Output,
+  DEFAULT_AGENT_CONFIGS_PACKAGES,
+} from '../../common';
+import { getPackageInfo } from './epm/packages';
+import { datasourceService } from './datasource';
 
 export async function setup(
   soClient: SavedObjectsClientContract,
   callCluster: CallESAsCurrentUser
 ) {
-  await Promise.all([
-    outputService.ensureDefaultOutput(soClient),
-    agentConfigService.ensureDefaultAgentConfig(soClient),
-
+  const [installedPackages, defaultOutput, config] = await Promise.all([
     // packages installed by default
     ensureInstalledDefaultPackages(soClient, callCluster),
+    outputService.ensureDefaultOutput(soClient),
+    agentConfigService.ensureDefaultAgentConfig(soClient),
   ]);
+
+  // ensure default packages are added to the default conifg
+  const configWithDatasource = await agentConfigService.get(soClient, config.id, true);
+  if (!configWithDatasource) {
+    throw new Error('Config not found');
+  }
+  if (
+    configWithDatasource.datasources.length &&
+    typeof configWithDatasource.datasources[0] === 'string'
+  ) {
+    throw new Error('Config not found');
+  }
+  for (const installedPackage of installedPackages) {
+    const packageShouldBeInstalled = !DEFAULT_AGENT_CONFIGS_PACKAGES.some(
```

Comment:
not sure i'm understanding this or finding the naming confusing.  if `packageShouldBeInstalled`, then do nothing?

Reply:
the name is totally confusing I should have named it `packageShouldBeInstalled` but I prefer your solution going to update

Reply:
thanks!